### ISSUE TITLE
[EVENT] deeplearning study group

### ISSUE THREAD

**Issue by marimeireles** [PyLadiesBerlin/community-organisation#9]
I'm proposing holding space for a study group on the dl course by hugging face (https://huggingface.co/deep-rl-course/unit1/introduction?fw=pt)
I need to have it discussed in a c-base circle
the circle happens next Tuesday 14.03
I'll be there also to join as a member and we might have cbase as a new location to host our events =)

**Comment by marimeireles** [comment]
Some updates: I've applied to c-base. Soon should get the reply that membership is okay.
Still not sure how to spin up this group. Should we have general objectives for the week? Or for every two weeks?
I was looking into [Spinning Up as a Deep RL Researcher](https://spinningup.openai.com/en/latest/spinningup/spinningup.html#id49) by openai and they have some good resources and what I consider good guidelines to go about building knowledge to understand DL.

Maybe we can decide on a theme for the week together, learn about it, show demos? Should I post recommendations of what to read and do? Not of specific project but broader themes, so we have something that tie us together. We can present to each other our demos. Or we can always elect someone in the week that will give everyone else a small presentation of what was the week's theme... Not sure, folks. 

Will update this again once I have a positive reply by cbase.

@linges @morganrivers Emma, feel free to drop any ideas.

**Comment by marimeireles** [comment]
Here's a rough outline:
- [introductory knowledge](https://spinningup.openai.com/en/latest/spinningup/spinningup.html#the-right-background), estimated time is variable here, depends on how much you know already
  - Build up a solid mathematical background. From probability and statistics, feel comfortable with random variables, Bayes’ theorem, chain rule of probability, expected values, standard deviations, and importance sampling. From multivariate calculus, understand gradients and (optionally, but it’ll help) Taylor series expansions
  - Build up a general knowledge of deep learning. You don’t need to know every single special trick and architecture, but the basics help. Know about standard architectures ([MLP](http://ufldl.stanford.edu/tutorial/supervised/MultiLayerNeuralNetworks/), [vanilla RNN](http://karpathy.github.io/2015/05/21/rnn-effectiveness/), [LSTM](https://arxiv.org/abs/1503.04069) ([also see this blog](http://colah.github.io/posts/2015-08-Understanding-LSTMs/)), [GRU](https://arxiv.org/abs/1412.3555v1), [conv](http://colah.github.io/posts/2014-07-Conv-Nets-Modular/) [layers](https://cs231n.github.io/convolutional-networks/), [resnets](https://arxiv.org/abs/1512.03385), [attention](https://arxiv.org/abs/1409.0473) [mechanisms](https://arxiv.org/abs/1706.03762)), common regularizers ([weight decay](https://papers.nips.cc/paper/563-a-simple-weight-decay-can-improve-generalization.pdf), [dropout](http://jmlr.org/papers/volume15/srivastava14a.old/srivastava14a.pdf)), normalization ([batch norm](https://arxiv.org/abs/1502.03167), [layer norm](https://arxiv.org/abs/1607.06450), [weight norm](https://arxiv.org/abs/1602.07868)), and optimizers ([SGD, momentum SGD](http://ufldl.stanford.edu/tutorial/supervised/OptimizationStochasticGradientDescent/), [Adam](https://arxiv.org/abs/1412.6980), [others](https://arxiv.org/abs/1609.04747)). Know what the [reparameterization trick](https://arxiv.org/abs/1312.6114) is. 
- we start with the [fast.ai course](https://course.fast.ai/), 44h of video class (?) + time for exercises, couldn't find estimates of how long
- implement algorithms, estimated time is around 200h
  - [Which algorithms?](https://spinningup.openai.com/en/latest/spinningup/spinningup.html#id51) You should probably start with vanilla policy gradient (also called REINFORCE), DQN, A2C (the synchronous version of A3C), PPO (the variant with the clipped objective), and DDPG, approximately in that order. The simplest versions of all of these can be written in just a few hundred lines of code (ballpark 250-300), and some of them even less (for example, a no-frills version of VPG can be written in about 80 lines). Write single-threaded code before you try writing parallelized versions of these algorithms. (Do try to parallelize at least one.)
And that's it!
Once we've done these things we'll probably know RL :)
Now I'm a member rat c-base and will propose to have a meetup space. Will let you know through here about possible dates.

**Comment by marimeireles** [comment]
This is ongoing! I'm stuck because of cbase has bi-monthly meetings and it's really hard to get a hold on them.

**Comment by marimeireles** [comment]
This is finally happening yaay!
Should I close this? Even though it's an ongoing event.

**Comment by berlin-pyladies** [comment]
closing now as for now at least it has stopped. Awesome work @marimeireles 🙌
